tests/e2e/configurer/chain/queries.go

```diff
@@ -20,12 +20,18 @@ import (
 	superfluidtypes "github.com/osmosis-labs/osmosis/v7/x/superfluid/types"
 )
 
-func (c *Config) QueryRPC(path string) ([]byte, error) {
-	var err error
+func (c *Config) QueryRPC(validatorIdx int, path string) ([]byte, error) {
+	// add the URL for the given validator ID, and pre-pend to to path.
+	hostPort, err := c.containerManager.GetValidatorHostPort(c.Id, validatorIdx, "1317/tcp")
+	require.NoError(c.t, err)
+	endpoint := fmt.Sprintf("http://%s", hostPort)
+	fullQueryPath := fmt.Sprintf("%s/%s", endpoint, path)
+
+	// TODO: Document why we have this loop
```

Comment:
We can have a `require.Eventually()` for this instead. Going to make an issue right away so we can remove
```suggestion
```

Reply:
https://github.com/osmosis-labs/osmosis/issues/2051